front/components/agent_builder/capabilities/mcp/MCPServerViewsDialog.tsx

```diff
@@ -0,0 +1,694 @@
+          description: `${mcpServerView.server.name} configuration has been updated.`,
+          type: "success",
+        });
+      } else {
+        // Add mode: update the selected tool with the configured action
+        setSelectedToolsInDialog((prev) => {
+          const existingToolIndex = prev.findIndex(
+            (tool) => tool.type === "MCP" && tool.view.sId === mcpServerView.sId
+          );
+
+          if (existingToolIndex >= 0) {
+            // Update existing tool with configuration
+            const updated = [...prev];
+            updated[existingToolIndex] = {
+              type: "MCP",
+              view: mcpServerView,
+              configuredAction,
+            };
+            return updated;
+          } else {
+            // Add new configured tool
+            return [
+              ...prev,
+              {
+                type: "MCP",
+                view: mcpServerView,
+                configuredAction,
+              },
+            ];
+          }
+        });
+
+        // Navigate back to tool selection page
+        setCurrentPageId(CONFIGURATION_DIALOG_PAGE_IDS.TOOL_SELECTION);
+        setConfigurationTool(null);
+        setConfigurationMCPServerView(null);
+      }
+    } catch (error) {
+      sendNotification({
+        title: "Configuration failed",
+        description:
+          "There was an error configuring the tool. Please try again.",
+        type: "error",
+      });
+    }
+  }, [
+    configurationTool,
+    form,
+    mcpServerView,
+    mode,
+    onActionUpdate,
+    sendNotification,
+    onModeChange,
+  ]);
+
+  const getFooterButtons = () => {
+    const isToolSelectionPage =
+      currentPageId === CONFIGURATION_DIALOG_PAGE_IDS.TOOL_SELECTION;
+    const isConfigurationPage =
+      currentPageId === CONFIGURATION_DIALOG_PAGE_IDS.CONFIGURATION;
+    const isInfoPage = currentPageId === CONFIGURATION_DIALOG_PAGE_IDS.INFO;
+
+    if (isToolSelectionPage) {
+      return {
+        leftButton: {
+          label: "Cancel",
+          variant: "outline" as const,
+          onClick: handleCancel,
+        },
+        rightButton: {
+          label:
+            selectedToolsInDialog.length > 0
+              ? `Add ${selectedToolsInDialog.length} tool${selectedToolsInDialog.length > 1 ? "s" : ""}`
+              : "Add tools",
+          variant: "primary",
+          disabled: selectedToolsInDialog.length === 0,
+          onClick: async () => {
+            await handleAddSelectedTools();
+            setIsOpen(false);
+          },
+        },
+      };
+    }
+
+    if (isConfigurationPage) {
+      if (isEditMode) {
+        // Edit mode: only Cancel and Save buttons
+        return {
+          leftButton: {
+            label: "Cancel",
+            variant: "outline",
+            onClick: handleCancel,
+          },
+          rightButton: {
+            label: "Save Changes",
+            variant: "primary",
+            onClick: handleConfigurationSave,
```

Comment:
Suggestion to valid and get access to the form data inside the save function
```suggestion
            onClick: form.handleSubmit(handleConfigurationSave),
```